fedimint-server/src/config/api.rs

```diff
@@ -1,775 +1,226 @@
-    /// which comes from the leader
-    fn get_config_gen_params(
-        &self,
-        request: &ConfigGenParamsRequest,
-        mut consensus: ConfigGenParamsConsensus,
-    ) -> ApiResult<ConfigGenParams> {
-        let local_connection = self.local_connection()?;
-
-        let (our_id, _) = consensus
-            .peers
-            .iter()
-            .find(|(_, param)| local_connection.tls_cert == param.cert)
-            .ok_or(ApiError::bad_request(
-                "Our TLS cert not found among peers".to_string(),
-            ))?;
-
-        let mut combined_params = vec![];
-        let default_params = self.settings.default_params.modules.clone();
-        let local_params = request.modules.clone();
-        let consensus_params = consensus.modules.clone();
-        // Use defaults in case local or consensus params are missing
-        for (id, kind, default) in default_params.iter_modules() {
-            let consensus = &consensus_params.get(id).unwrap_or(default).consensus;
-            let local = &local_params.get(id).unwrap_or(default).local;
-            let combined = ConfigGenModuleParams::new(local.clone(), consensus.clone());
-            // Check that the params are parseable
-            let module = self.settings.registry.get(kind).expect("Module exists");
-            module.validate_params(&combined).map_err(|e| {
-                ApiError::bad_request(format!(
-                    "Module {} params invalid: {}",
-                    id,
-                    itertools::join(e.chain(), ": ")
-                ))
-            })?;
-            combined_params.push((id, kind.clone(), combined));
-        }
-        consensus.modules = ServerModuleConfigGenParamsRegistry::from_iter(combined_params);
-
-        let local = ConfigGenParamsLocal {
-            our_id: *our_id,
-            our_private_key: local_connection.tls_private,
-            api_auth: self.auth()?,
-            p2p_bind: self.settings.p2p_bind,
-            api_bind: self.settings.api_bind,
-            max_connections: self.settings.max_connections,
+        let our_id = state
+            .connection_info
+            .keys()
+            .position(|url| *url == self.settings.api_url)
+            .expect("We inserted the key above.");
+
+        let params = ConfigGenParams {
+            local: ConfigGenParamsLocal {
+                our_id: PeerId::from(our_id as u16),
+                our_private_key: local_params.tls_private,
+                api_auth: local_params.auth,
+                p2p_bind: self.settings.p2p_bind,
+                api_bind: self.settings.api_bind,
+            },
+            consensus: ConfigGenParamsConsensus {
+                peers: (0..)
+                    .map(|i| PeerId::from(i as u16))
+                    .zip(state.connection_info.values().cloned())
+                    .collect(),
+                meta: BTreeMap::from_iter(vec![("federation_name".to_string(), federation_name)]),
```

Comment:
We should drop initializing this field eventually once there's enough ecosystem support for the meta module (I still have to patch it into Fedimint Observer iirc :grimacing: )

Reply:
It isn't, it can be overwritten xD